koocook_core/models/base.py

```diff
@@ -31,7 +31,10 @@ def as_json(self) -> str:
 class ModelEncoder(JSONEncoder):
     def default(self, obj: models.Model):
         if hasattr(obj, 'as_dict'):
-            return obj.as_dict()
+            try:
+                return obj.as_dict()
+            except TypeError:
+                return obj.as_dict()
         else:
             if isinstance(obj, models.Model):
                 if isinstance(obj, SerialisableModel):
```

Comment:
How does this work? Wouldn't doing the same statement raise the same error?

Reply:
Fixed. Thanks!